src/Common/Integrations/Plugin_Merge_Provider_Abstract.php

```diff
@@ -0,0 +1,326 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	protected function did_update_to_merge_version(): bool {
+		return version_compare( Tribe__Settings_Manager::get_option( $this->get_last_version_option_key() ), $this->get_merged_version(), '<' );
+	}
+
+	/**
+	 * Is the child plugin active?
+	 *
+	 * @since TBD
+	 */
+	protected function is_child_plugin_active(): bool {
+		return is_plugin_active( $this->get_plugin_file_key() );
+	}
+
+	/**
+	 * Initializes the merged compatibility checks.
+	 *
+	 * @since TBD
+	 */
+	public function init() {
+		// Load our is_plugin_activated function.
+		if ( ! function_exists( 'is_plugin_activated' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		if ( $this->updated_to_merge_version && ! $this->is_child_plugin_active() ) {
+			// Leave a notice of the recent update.
+			$this->send_updated_notice();
+			$this->init_merged_plugin();
+			return;
+		}
+
+		// If the Event Ticket Wallet Plus plugin is active, we need to deactivate it before continuing to avoid a fatal.
+		if ( $this->is_child_plugin_active() ) {
+			$this->deactivate_plugin();
+
+			// Leave a notice of the forced deactivation.
+			$this->send_updated_merge_notice();
+			return;
+		}
+
+		// If the action is to activate the plugin, we should not continue to avoid a fatal.
+		if ( $this->is_activating_plugin() ) {
+			add_action( 'activated_plugin', [ $this, 'deactivate_plugin' ] );
+
+			// Remove "Plugin activated" notice from redirect.
+			add_action( 'activate_plugin', [ $this, 'remove_activated_from_redirect' ] );
+			// Leave a notice of the forced deactivation.
+			$this->send_activating_merge_notice();
+			return;
+		}
+
+		// If the plugin is not active and the action is not to activate it, we can proceed with the merge.
+		$this->init_merged_plugin();
+	}
+
+	/**
+	 * If any initialization is necessary for the merged plugin to work after child plugin deactivation is resolved.
+	 *
+	 * @since TBD
+	 */
+	public function init_merged_plugin() {
+		// Implement the merged plugin initialization.
+	}
+
+	/**
+	 * Deactivates the merged plugin.
+	 *
+	 * @since TBD
+	 */
+	public function deactivate_plugin() {
+		deactivate_plugins( $this->get_plugin_file_key() );
+	}
+
+	/**
+	 * Adds the hook to remove the "Plugin activated" notice from the redirect.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $plugin The plugin file path.
+	 */
+	public function remove_activated_from_redirect( $plugin ) {
+		if ( basename( $plugin ) === basename( $this->get_plugin_file_key() ) ) {
+			add_filter( 'wp_redirect', [ $this, 'filter_remove_activated_from_redirect' ] );
+		}
+	}
+
+	/**
+	 * Filter the redirect location to remove the "activate" query arg.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $location The redirect location.
+	 *
+	 * @return string The redirect location without the "activate" query arg.
+	 */
+	public function filter_remove_activated_from_redirect( $location ) {
```

Comment:
```suggestion
	public function filter_remove_activated_from_redirect( $location ): string {
```

Reply:
I'd personally throw `void` on the public (thus overridable) ones that are never meant to return a value as well - to enforce that behavior.